src/Documentation/Grain-Versioning/Backward-compatibility-guidelines.md

```diff
@@ -0,0 +1,122 @@
+# Backward compatibility guidelines
+
+Writing backward compatible code can be hard and difficult to test.
+
+## Never change the signature of existing methods
+
+Because of the way on how Orleans serializer work, you should never change the signature
+of existing methods.
+
+The following example is correct:
+
+``` cs
+[Version(1)]
+public interface IMyGrain : IGrainWithIntegerKey
+{
+  // First method
+  Task MyMethod(int arg);
+}
+```
+``` cs
+[Version(2)]
+public interface IMyGrain : IGrainWithIntegerKey
+{
+  // Method inherited from V1
+  Task MyMethod(int arg);
+
+  // New method added in V2
+  Task MyNewMethodMethod(int arg, obj o);
```

Comment:
nit: `Method` twice in the name is odd. I see it in many places, so it might have been by design, but I don't really understand it.